todoapp/app/src/main/res/layout/tasks_frag.xml

```diff
@@ -13,26 +13,29 @@
   ~ See the License for the specific language governing permissions and
   ~ limitations under the License.
   -->
-<layout xmlns:android="http://schemas.android.com/apk/res/android">
+<layout xmlns:android="http://schemas.android.com/apk/res/android"
+        xmlns:app="http://schemas.android.com/apk/res-auto">
 
     <data>
 
         <import type="android.view.View" />
 
         <variable
-            name="tasks"
-            type="com.example.android.architecture.blueprints.todoapp.tasks.TasksViewModel" />
+            name="view"
+            type="com.example.android.architecture.blueprints.todoapp.tasks.TasksFragment" />
 
         <variable
-            name="actionHandler"
-            type="com.example.android.architecture.blueprints.todoapp.tasks.TasksContract.Presenter" />
+            name="viewmodel"
+            type="com.example.android.architecture.blueprints.todoapp.tasks.TasksViewModel" />
+
     </data>
 
-<com.example.android.architecture.blueprints.todoapp.tasks.ScrollChildSwipeRefreshLayout
+<com.example.android.architecture.blueprints.todoapp.ScrollChildSwipeRefreshLayout
     android:id="@+id/refresh_layout"
     android:layout_width="match_parent"
     android:layout_height="match_parent"
-    android:onRefresh="@{actionHandler}">
+    android:onRefresh="@{viewmodel}"
```

Comment:
That would be making the view (in this case the layout) a bit smarter. To keep it dumb, I prefer to have that call to loadTasks in the ViewModel.